src/Services/API/Project/ProjectService.API/Controllers/ProjectController.cs

```diff
@@ -0,0 +1,131 @@
+﻿using ProjectService.Contract.DataTransfer;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Swashbuckle.AspNetCore.Annotations;
+using EnterpriseAssistant.DataAccess;
+using Microsoft.EntityFrameworkCore;
+using EnterpriseAssistant.DataAccess.Entities;
+using MediatR;
+using Mapster;
+using DepartmentService.Contract.Commands;
+using DepartmentService.Contract.DataTransfer;
+using EnterpriseAssistant.Application.Shared;
+
+namespace ProjectService.API.Controllers;
+
+[AllowAnonymous]
+[ApiController]
+[Route("api/project")]
+public class ProjectController : ControllerBase
+{
+    IMediator _mediator;
+    private readonly EnterpriseAssistantDbContext _context;
+
+    public ProjectController(EnterpriseAssistantDbContext context, IMediator mediator)
+    {
+        _context = context;
+        _mediator = mediator;
+    }
+
+    /*
+    [HttpPost]
+    [SwaggerOperation(Summary ="Create project", Description = "Create a project")]
+
+    public async Task<ActionResult<ProjectCreateDto>> CreateProject([FromBody] ProjectCreateDto project)
+    {
+        var result = await _mediator.Send(new CreateProject(project));
+        return Ok(result);
+    }
+
+
+    [HttpGet("{projectId:long}")]
+    [SwaggerOperation(Summary = "Get prject by Id", Description = "Get project")]
+    public async Task<ActionResult<IEnumerable<ProjectDto>>> GetProjectById(
+        [Range(1, long.MaxValue), FromRoute] long projectId)
+    {
+        var result = await _mediator.Send(new GetProjectById(projectId));
+
+        return result.Match<ActionResult>(Ok, x => NotFound());
+    }
+
+    [HttpDelete("{projectId:long}")]
+     [SwaggerOperation(Summary = "Delete project by id", Description = "delete project")]
+      public async Task<IActionResult> ProjectDelete(long id)
+     {
+
+     }
+    */
+
+    [HttpPost]
+    [SwaggerOperation(Summary = "Add project", Description = "add project")]
+    public async Task<ActionResult<Project>> CreateProject(ProjectCreateDto request)
+    {
+        var response = await _mediator.Send(new CreateDepartment(request.DepartmentCreate, User.GetAuthContext()));
+        if (response.IsT1)
+        {
+            return BadRequest(response.AsT1.Message);
+        }
+        var project = request.Adapt<Project>();
+        _context.Projects.Add(project);
+        await _context.SaveChangesAsync();
+        return CreatedAtAction(nameof(CreateProject), new { id = project.Id }, project);
+    }
+
+    [HttpDelete("{id}")]
+    [SwaggerOperation(Summary = "Delete project by id", Description = "delete project")]
+    public async Task<IActionResult> DeleteProject(long id)
+    {
+        var project = await _context.Projects.FindAsync(id);
+        if (project == null)
+        {
+            return NotFound();
+        }
+
+        project.IsSoftDeleted = true;
```

Comment:
![image](https://user-images.githubusercontent.com/46058617/172420212-f3517c7e-f08d-4e9d-b7f5-06a8ec669754.png)
for updating one property use this, and Update for update the entire entity